panels/updates/cc-updates-panel.c

```diff
@@ -285,52 +319,100 @@ get_network_status_icon_name (CcUpdatesPanel *self,
 }
 
 static void
-update_active_network (CcUpdatesPanel *self)
+load_tariff_from_connection (CcUpdatesPanel *self,
+                             NMConnection   *connection)
 {
-  NMAccessPoint *ap;
-  NMConnection *connection;
-  NMDevice *device;
-  const gchar *icon_name;
+  g_autoptr(MwtTariff) tariff = NULL;
+  g_autoptr(GError) error = NULL;
+  const gchar *tariff_value = NULL;
+  gboolean tariff_enabled = FALSE;
 
-  get_active_connection_and_device (self, &device, &connection, &ap);
+  g_debug ("Loading tariff from connection");
 
-  /* Setup the new device */
-  if (g_set_object (&self->current_device, device) && device)
+  if (connection)
     {
-      /* Cleanup previous device handler */
-      if (self->changed_id > 0)
+      NMSettingUser *setting_user;
+      const gchar *tariff_enabled_value = NULL;
+
+      setting_user = NM_SETTING_USER (nm_connection_get_setting (connection, NM_TYPE_SETTING_USER));
```

Comment:
`nm_connection_get_setting()` can apparently return `NULL`. Can that ever happen for `NM_TYPE_SETTING_USER`? If so, the null value needs to be handled gracefully. If not, please add a `g_assert (setting_user != NULL);`.